assignments/this.js

```diff
@@ -1,26 +1,69 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. -Window/ global Object binding
+    - Here `this` refers to the window object, an object which has a global scope in a given context
+
+* 2. Implicit Binding
+    - When `this` is used inside an object, the binding is said to be implicit
+
+* 3. New Binding
+     - here `this` refers to the instance of the object returned by a constructor function
+
+* 4. - Explicit Binding
+     `this` here refers to a binding that occurs when `apply` and `call` methods are used
```

Comment:
- Most functions in JS have .call() and .apply() methods available to them. Both of these methods take an object
as their first parameter on which 'this' should be used. This is how you explicitly say that you want 'this' to be the object passed in the parameter.